op-batcher/batcher/driver.go

```diff
@@ -284,86 +287,130 @@ func (l *BatchSubmitter) calculateL2BlockRangeToStore(ctx context.Context) (eth.
 func (l *BatchSubmitter) loop() {
 	defer l.wg.Done()
 
-	ticker := time.NewTicker(l.PollInterval)
-	defer ticker.Stop()
+	loadTicker := time.NewTicker(l.PollInterval)
+	defer loadTicker.Stop()
+	publishTicker := time.NewTicker(100 * time.Millisecond)
+	defer publishTicker.Stop()
+
+	receiptsCh := make(chan txmgr.TxReceipt[txData])
+	queue := txmgr.NewQueue[txData](l.txMgr, l.MaxPendingTransactions, l.metr.RecordPendingTx)
+
 	for {
 		select {
-		case <-ticker.C:
+		case <-loadTicker.C:
 			l.loadBlocksIntoState(l.shutdownCtx)
-			l.publishStateToL1(l.killCtx)
+		case <-publishTicker.C:
+			_, _ = queue.TrySend(l.killCtx, l.publishStateToL1Factory(), receiptsCh)
+		case r := <-receiptsCh:
+			l.handleReceipt(r)
```

Comment:
Is it necessary to decouple tx sending from block loading? Was it added to load the L1Tip only at the time of tx creation? I guess that's also the reason why there's lazy-generation of TxCandidates with a factory?

Could we instead, similarly to before, iterate over `l.state.TxData` and then call `queue.Send`? I think it's also fine to block if the queue is full, we wouldn't gain anything from loading L2 blocks earlier anyways. If that's possible, it could simplify a few things, like not needing to work with an injected factory or `TrySend`.

Reply:
The original reason was to ensure that L1Tip was up-to-date, but because this is only done for channel duration timeouts, if the channel manager returns a tx to send it is either because the channel already timed out or was full, so it's fine for the L1 tip to be a little behind in this case.

I refactored to remove the factory and just use `queue.Send`.